pom.xml

```diff
@@ -964,6 +964,19 @@
         <version>0.8.1</version>
         <scope>test</scope>
       </dependency>
+      <dependency>
+        <groupId>com.google.guava</groupId>
+        <!-- https://javadoc.io/doc/com.google.guava/guava-testlib/latest/com/google/common/testing/TestLogHandler.html used in log4j-to-jul tests -->
+        <artifactId>guava-testlib</artifactId>
+        <version>31.0.1-jre</version>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>com.google.truth</groupId>
+        <artifactId>truth</artifactId>
+        <version>1.1.3</version>
+        <scope>test</scope>
+      </dependency>
```

Comment:
I agree, it gives us a touch more flexibility. OTOH, if a new class does not declare protected methods, does it matter?

Reply:
Ah, right. So I personally really like https://truth.dev and assumed it was a "more modern" alternative to https://assertj.github.io/doc/... but it seems like AssertJ may have "caught up" since I last had a closer look at it (years ago) and also has that "fluent style" that I like in Truth (only now, or always had?). There is an internal process at work that is a tiny bit easier for me if I can use Truth instead of AssertJ, so I have to ask 😸 if this feedback is a "suggestion" or a "requirement", for merge? I totally do understand if you make it a hard requirement, for consistency across Log4j, and am happy to switch and adapt, but I thought I would ask just to double check once before doing so. 

> I agree, it gives us a touch more flexibility.

Which one, AssertJ or Truth?

> OTOH, if a new class does not declare protected methods, does it matter?

Sorry I did not understand this (my choice of assertion framework here wasn't related to testing declared protected methods).

Reply:
I think Gary meant to reply to https://github.com/apache/logging-log4j2/pull/653#discussion_r775083886

I haven't used the Truth library outside of perhaps a couple bug-fixes in guava/error-prone. I will be able to review much more quickly and easily if you use frameworks I'm familiar with. Truth appears to bring in a few large transitive dependencies as well. Unless we're consolidating all the assertj/junit asserts into truth, we're probably better off avoiding  [the lava layer](http://mikehadlow.blogspot.com/2014/12/the-lava-layer-anti-pattern.html).

Reply:
ACK, then I'll rework this to use AssertJ instead of Truth, no worries. (FYI I've added additional tests just now.) Expected timeframe: Coming couple of days, hopefully some time this week (but not today anymore and unlikely tomorrow).

Reply:
Huh - AssertJ and Truth are 99% (`isSameInstanceAs` => `isSameAs`) API compatible! COOL. Done.